File: app/schemas/therapist.py
from pydantic import BaseModel
from typing import Optional


class TherapistCreate(BaseModel):
    full_name: str
    specialization: str
    bio: Optional[str] = None
    languages: Optional[str] = None
    city: Optional[str] = None
    is_online: bool = True
    price_per_session: Optional[float] = None


class TherapistResponse(TherapistCreate):
    id: int

    class Config:
        from_attributes = True

File: frontend/src/types/therapist.ts
export type Therapist = {
  user_id: number
  full_name: string
  specialization: string
  bio?: string | null
  /** Backend currently returns a single string (nullable) */
  languages?: string | null
  city?: string | null
  is_online: boolean
  /** Backend returns Decimal; it may arrive as number or string */
  price_per_session?: number | string | null

  /** Frontend-only optional field (backend may add later) */
  image_url?: string | null
}

export type TherapistSearchParams = {
  city?: string
  specialization?: string
  min_price?: number
  max_price?: number
  is_online?: boolean
  limit?: number
  offset?: number
}

export type TherapistUpdateBody = {
  full_name?: string | null
  specialization?: string | null
  bio?: string | null
  languages?: string | null
  city?: string | null
  is_online?: boolean | null
  price_per_session?: number | null
}
